### ISSUE TITLE
Training script for T2I adapter bug?

### ISSUE THREAD

**Issue by darshats** [huggingface/diffusers#5403]
Hi,
The training script for t2i adapter also sets the unet for train:

https://github.com/huggingface/diffusers/blob/07b297e7dec3f475a880c15addcb6c02c0690992/examples/t2i_adapter/train_t2i_adapter_sdxl.py#L900C5-L900C9

Is this by design or a bug? My understanding is that only adapter is trained and unet does not need to be finetuned. The inference scripts also do not take a trained unet.

thanks
Darshat

**Comment by sayakpaul** [comment]
It is not a training bug. @patil-suraj will be able to shed more light here.

**Comment by darshats** [comment]
@patil-suraj  - would appreciate if you can provide some insight on how to proceed here? Is training the unet needed to train an adapter?

**Comment by sayakpaul** [comment]
Cc: @MC-E 

Yes, this is necessary, because the gradient needs to be backpropagated to the adapter, so the gradient of UNet must be enabled. But the UNet parameters aren't updated as they aren't passed to the optimizer.
